### PR TITLE
new example plugin schema

### PR DESCRIPTION
**Pull request by DrMegavolt**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
   "author": "",
-  "license": "Apache-2.0"
+  "license": "Apache-2.0",
+  "dependencies": {},
+  "devDependencies": {
+    "eslint": "^4.9.0",
+    "eslint-config-standard": "^10.2.1",
+    "eslint-plugin-import": "^2.8.0",
+    "eslint-plugin-node": "^5.2.1",
+    "eslint-plugin-promise": "^3.6.0"
```
Comment (XVincentX) [inline package.json]: `eslint-plugin-node` and `eslint-plugin-promise` are being installed, but aren't imported in .eslintrc
You should add those to the `plugin` section of that file (note — `plugin-import` is required by `config-standard`)

File: .gitingore [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+.vscode
+.idea
+node_modules/
+npm-debug.log
+example/keys/*.pem
+.DS_Store
+**/.DS_Store
+coverage
+.nyc_output
+coverage.lcov
```
Comment (XVincentX) [inline .gitingore]: Nitpick: A lot of these files aren't here at all (such as `dist, .vscode, .idea`).

File: policies/example-policy.js [hunk 3]
```diff
@@ -2,8 +2,15 @@ module.exports = {
   name: 'example',
   policy: (actionParams) => {
     return (req, res, next) => {
-      // eslint-disable-next-line no-console
       console.log('executing policy-from-example-plugin with params', actionParams);
     };
+  },
+  schema: {  // This is for Admin API to validate params
+    url: {
```
Comment (XVincentX) [inline policies/example-policy.js]: I'm afraid this is not a valid schema. It should be:

```javascript
{
    type: "object",
    properties: {
        url: {
            title: 'url',
            description: 'the url to initialize',
            type: 'string',
            required: true
        }

    }
}
```

File: policies/advanced-policy.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+module.exports = pluginContext => {
+  name: 'exampleWithContext',
+  policy: (actionParams) => {
+    return (req, res, next) => {
+      // eslint-disable-next-line no-console
+      console.log('executing policy-from-example-plugin with params', actionParams);
+    };
+  },
+  schema: {  // This is for Admin API to ask about params
+    url: {
```
Comment (XVincentX) [inline policies/advanced-policy.js]: Invalid JSON schema

**Review by XVincentX — CHANGES_REQUESTED** [review]
Couple of things to change, the schema in particular is invalid.

File: .gitingore (hunk 2 above)
Comment (DrMegavolt) [reply .gitingore]: copy paste from EG

**Comment by DrMegavolt** [comment]
@XVincentX updated

**Review by XVincentX — APPROVED** [review]

File: policies/example-policy.js (hunk 3 above)
Comment (xujiyuan) [reply policies/example-policy.js]: $id is required now?

Comment (XVincentX) [reply policies/example-policy.js]: It has always been required as far as I remember

**Pull request merged**